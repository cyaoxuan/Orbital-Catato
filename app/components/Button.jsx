import { View } from "react-native";
import { Button } from "react-native-paper";

const PillButton = (props) => {
    return (
        <View style={{ flexDirection: "row" }}>
                <Button style = {{ margin: 10, width: props.width }}
                    disabled={props.disabled}
                    mode={props.mode}
                    onPress={props.onPress}>
                        {props.label}
                </Button>
        </View>
    )
}

export { PillButton };