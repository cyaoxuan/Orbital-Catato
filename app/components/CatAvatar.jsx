import { StyleSheet, TouchableOpacity, View } from "react-native";
import { Avatar } from "react-native-paper";
import { BodyText, TitleText } from "./Text";

// Cat Avatar with circular image and name
// props: photoURL (string), size (image size || number), variant (font size || string), name (string)
const CatAvatar = (props) => {
    return (
        <View style={styles.avatarContainer}>
            <Avatar.Image
                testID="avatar-image"
                source={
                    props.photoURL
                        ? { uri: props.photoURL }
                        : require("../../assets/placeholder.png")
                }
                size={props.size || 100}
            />
            <BodyText variant={props.variant} text={props.name || "Name"} />
        </View>
    );
};

// Cat Avatar with circular image and name, touchable
// props: photoURL (string), size (image size || number), variant (font size || string), name (string), onPress (callback)
const TouchableCatAvatar = (props) => {
    return (
        <TouchableOpacity
            testID="cat-avatar"
            style={styles.avatarContainer}
            activeOpacity={0.7}
            onPress={props.onPress}
        >
            <Avatar.Image
                testID="avatar-image"
                source={
                    props.photoURL
                        ? { uri: props.photoURL }
                        : require("../../assets/placeholder.png")
                }
                size={props.size || 100}
            />
            <BodyText variant={props.variant} text={props.name || "Name"} />
        </TouchableOpacity>
    );
};

export { CatAvatar, TouchableCatAvatar };

const styles = StyleSheet.create({
    avatarContainer: {
        alignItems: "center",
    },
});
