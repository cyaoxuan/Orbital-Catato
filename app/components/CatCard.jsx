import { Card } from "react-native-paper";
import { IconTextField } from "./InfoText";

const CatCard = (props) => {
    return (
        <Card style={{ height: 280, width: 280, margin: 4 }} 
            onPress={() => {}}
            mode="elevated">
            <Card.Cover style={{ height: 140, width: 280, resizeMode: "cover"}} 
                source={props.catImage} 
            />
            <Card.Title title={props.catName} titleVariant="headlineMedium"/>
            <Card.Content style={{ paddingBottom: 0, paddingHorizontal: 4 }}>
                <IconTextField 
                    iconName={props.iconName1} 
                    iconSize={24} 
                    variant="titleLarge"
                    field={props.field1}
                    info={props.info1}
                />
                <IconTextField 
                    iconName={props.iconName2} 
                    iconSize={24} 
                    variant="titleLarge" 
                    field={props.field2}
                    info={props.info2}
                />
            </Card.Content>
        </Card>
    );
};

export { CatCard };