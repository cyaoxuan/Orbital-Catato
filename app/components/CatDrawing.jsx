import { Image } from "react-native";

export const PantingCat = ({ size }) => {
    return (
        <Image
            source={require("../../assets/drawing-cats/cat-drawing-1.png")}
            style={{ height: size, width: size }}
        />
    );
};

export const PopCat = ({ size }) => {
    return (
        <Image
            source={require("../../assets/drawing-cats/cat-drawing-2.png")}
            style={{ height: size, width: size }}
        />
    );
};

export const SmilingGingerCat = ({ size }) => {
    return (
        <Image
            source={require("../../assets/drawing-cats/cat-drawing-3.png")}
            style={{ height: size, width: size }}
        />
    );
};

export const LoadingCat = ({ size }) => {
    return (
        <Image
            source={require("../../assets/drawing-cats/cat-drawing-4.png")}
            style={{ height: size, width: size }}
        />
    );
};

export const MaxwellCat = ({ size }) => {
    return (
        <Image
            source={require("../../assets/drawing-cats/cat-drawing-5.png")}
            style={{ height: size, width: size }}
        />
    );
};

export const SaladSmudgeCat = ({ size }) => {
    return (
        <Image
            source={require("../../assets/drawing-cats/cat-drawing-6.png")}
            style={{ height: size, width: size }}
        />
    );
};

export const StandingCat = ({ size }) => {
    return (
        <Image
            source={require("../../assets/drawing-cats/cat-drawing-7.png")}
            style={{ height: size, width: size }}
        />
    );
};
