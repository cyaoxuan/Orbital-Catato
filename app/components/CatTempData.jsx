// Temp data to showcase static pages

const catViewProps = [{
    iconName1: "information-circle",
    field1: "Status: ",
    iconName2: "location",
    field2: "Last Seen: "
}, {
    iconName1: "time",
    field1: "Last Fed: ",
    iconName2: "location",
    field2: "Last Seen: "
}];

const concernCatCards = [{
    catImage: require("../../assets/temp-cat.jpg"),
    catName: "Kitty",
    iconName1: catViewProps[0].iconName1,
    field1: catViewProps[0].field1,
    info1: "Injured",
    iconName2: catViewProps[0].iconName2,
    field2: catViewProps[0].field2,
    info2: "Engineering"
}, {
    catImage: require("../../assets/temp-cat.jpg"),
    catName: "Skitty",
    iconName1: catViewProps[0].iconName1,
    field1: catViewProps[0].field1,
    info1: "New",
    iconName2: catViewProps[0].iconName2,
    field2: catViewProps[0].field2,
    info2: "Utown"
}, {
    catImage: require("../../assets/temp-cat.jpg"),
    catName: "Mitty",
    iconName1: catViewProps[0].iconName1,
    field1: catViewProps[0].field1,
    info1: "Missing",
    iconName2: catViewProps[0].iconName2,
    field2: catViewProps[0].field2,
    info2: "Science"
}, {
    catImage: require("../../assets/temp-cat.jpg"),
    catName: "Bitty",
    iconName1: catViewProps[0].iconName1,
    field1: catViewProps[0].field1,
    info1: "Injured",
    iconName2: catViewProps[0].iconName2,
    field2: catViewProps[0].field2,
    info2: "Arts"
}];

const fedCatCards = [{
    catImage: require("../../assets/temp-cat.jpg"),
    catName: "Kitty",
    iconName1: catViewProps[1].iconName1,
    field1: catViewProps[1].field1,
    info1: "19/05 12:01",
    iconName2: catViewProps[1].iconName2,
    field2: catViewProps[1].field2,
    info2: "Engineering"
}];

export { concernCatCards, fedCatCards }