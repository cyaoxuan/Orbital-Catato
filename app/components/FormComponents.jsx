import { useState } from "react";
import { View, StyleSheet } from "react-native";
import {
    Button,
    HelperText,
    Menu,
    RadioButton,
    Text,
    TextInput,
} from "react-native-paper";
import { SelectList } from "react-native-dropdown-select-list";
import NumericInput from "react-native-numeric-input";
import RNDateTimePicker from "@react-native-community/datetimepicker";
import { timeOptions } from "../data/DateTimeOptions";

const titleVariant = "titleMedium";
const bodyVariant = "bodyMedium";

// For Location
const DropdownList = (props) => {
    const [selected, setSelected] = useState("");

    return (
        <View style={styles.container}>
            <Text variant={titleVariant}>{props.titleText}</Text>
            <SelectList
                setSelected={
                    props.setSelected
                        ? props.setSelected
                        : (val) => setSelected(val)
                }
                data={props.data || [{ key: "1", value: "value" }]}
                save="value"
            />
        </View>
    );
};

// Text inputs, description reasons, name, features etc
// Required field
const FormInput = (props) => {
    const [value, setValue] = useState("");

    return (
        <View style={styles.container}>
            <TextInput
                style={styles.formInput}
                disabled={props.disabled}
                multiline={props.multiline}
                label={props.label || "Label"}
                placeholder={props.placeholder}
                value={props.value ? props.value : value}
                onChangeText={
                    props.onChangeText ? props.onChangeText : setValue
                }
                error={props.value ? !props.value.trim() : !value.trim()}
            />
            <HelperText
                type="error"
                padding="none"
                visible={props.value ? !props.value.trim() : !value.trim()}
            >
                Error: {props.errorText || "Error"}
            </HelperText>
        </View>
    );
};

// For year
const NumberSpinner = (props) => {
    const [number, setNumber] = useState(new Date().getFullYear());

    return (
        <View style={styles.container}>
            <Text variant={titleVariant}>{props.titleText}</Text>
            <NumericInput
                type="up-down"
                rounded
                editable={false}
                minValue={props.min}
                maxValue={props.max}
                value={props.value ? props.value : number}
                onChange={
                    props.onChange
                        ? props.onChange
                        : (num) => {
                              setNumber(num);
                          }
                }
                totalWidth={120}
                totalHeight={40}
            />
        </View>
    );
};

// For time
const TimeInput = (props) => {
    const today = new Date();
    const [date, setDate] = useState(new Date());
    const [show, setShow] = useState(false);

    const onChange = (event, selectedDate) => {
        const currentDate = selectedDate;
        setShow(false);
        setDate(currentDate);
    };

    return (
        <View style={styles.container}>
            <Text variant={titleVariant}>{props.titleText}</Text>
            <Button
                mode="contained-tonal"
                onPress={props.onPress ? props.onPress : setShow}
                style={{ width: "50%", margin: 4 }}
            >
                Pick Time{" "}
            </Button>
            <Text variant={bodyVariant}>
                Selected Time:{" "}
                {props.displayTime
                    ? props.displayTime.toLocaleTimeString("en-GB", timeOptions)
                    : date.toLocaleTimeString("en-GB", timeOptions)}
            </Text>
            {(props.displayTime ? props.displayTime > today : date > today) && (
                <Text variant={bodyVariant} style={{ color: "crimson" }}>
                    Error: Selected future time! Are you a time traveller?
                </Text>
            )}
            {(props.show ? props.show : show) && (
                <RNDateTimePicker
                    mode="time"
                    display="spinner"
                    is24Hour={true}
                    value={props.value ? props.value : date}
                    onChange={props.onChange ? props.onChange : onChange}
                />
            )}
        </View>
    );
};

// For gender, sterilised, concerns, foster
const TwoRadioInput = (props) => {
    const [selected, setSelected] = useState("First");

    return (
        <View style={styles.container}>
            <Text variant={titleVariant}>{props.titleText}</Text>
            <RadioButton.Group
                value={props.value ? props.value : selected}
                onValueChange={
                    props.onValueChange
                        ? props.onValueChange
                        : (value) => setSelected(value)
                }
            >
                <View style={styles.radioButtonContainer}>
                    <Text variant={bodyVariant}>{props.firstText}</Text>
                    <RadioButton value={props.firstValue || "First"} />
                </View>
                <View style={styles.radioButtonContainer}>
                    <Text variant={bodyVariant}>{props.secondText}</Text>
                    <RadioButton value={props.secondValue || "Second"} />
                </View>
            </RadioButton.Group>
        </View>
    );
};

// To upload photos in form
const UploadPhotos = (props) => {
    const [visible, setVisible] = useState(false);
    const openMenu = () => setVisible(true);
    const closeMenu = () => setVisible(false);

    return (
        <View style={styles.container}>
            <Text variant={titleVariant}>Upload Photos:</Text>
            <View>
                <Menu
                    style={{ marginTop: -50 }}
                    visible={visible}
                    onDismiss={closeMenu}
                    anchor={
                        <Button
                            mode="contained-tonal"
                            icon="camera"
                            onPress={openMenu}
                            style={{ width: "50%", margin: 4 }}
                        >
                            Upload{" "}
                        </Button>
                    }
                >
                    <Menu.Item
                        leadingIcon="camera"
                        onPress={props.cameraOnPress}
                        title="Camera"
                    />
                    <Menu.Item
                        leadingIcon="image"
                        onPress={props.galleryOnPress}
                        title="Gallery"
                    />
                </Menu>
            </View>
            <Text variant={bodyVariant}>
                Image: {props.photoURI ? props.photoURI : ""}
            </Text>
        </View>
    );
};

export {
    DropdownList,
    FormInput,
    NumberSpinner,
    TimeInput,
    TwoRadioInput,
    UploadPhotos,
};

const styles = StyleSheet.create({
    formInput: {
        fontSize: 16,
        width: "100%",
        backgroundColor: "transparent",
        paddingHorizontal: 0,
        alignContent: "center",
    },

    container: {
        width: "90%",
        margin: 4,
    },

    radioButtonContainer: {
        flexDirection: "row",
        alignItems: "center",
        justifyContent: "space-between",
        paddingHorizontal: 4,
    },

    icon: {
        marginHorizontal: 8,
    },
});
