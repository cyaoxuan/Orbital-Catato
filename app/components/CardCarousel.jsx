import { View, FlatList } from "react-native";
import { CatCard } from "../components/CatCard";

const boxWidth = 300;

const Carousel = (props) => {
    return (
        <FlatList style={{ height: 300 }}
                horizontal
                snapToAlignment="center"
                decelerationRate="normal"
                showsHorizontalScrollIndicator={false}
                scrollEventThrottle={10}
                snapToInterval={boxWidth}
                contentOffset={{ x: -120, y: 0 }}
                ItemSeparatorComponent={() => <View style={{width: 20}} />}
                
                data={props.cards}
                renderItem={({item, index}) => {
                    return (
                        <View key={index}> 
                            <CatCard 
                                catImage={item.catImage}
                                catName={item.catName}
                                iconName1={item.iconName1}
                                field1={item.field1}
                                info1={item.info1}
                                iconName2={item.iconName2}
                                field2={item.field2}
                                info2={item.info2}
                            />
                        </View>
                        
                    );
                }}
        />

    );
};

export { Carousel }