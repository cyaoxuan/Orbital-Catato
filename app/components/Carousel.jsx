import { FlatList, View } from "react-native";
import { useNavigation } from "expo-router";
import { CatCard } from "./CatCard";
import { formatLastFed, formatLastSeenSimple } from "../utils/formatDetails";

function getInfo1(carouselType, cat) {
    if (!cat || !carouselType) {
        return "Unknown";
    }

    if (carouselType === "concern" && cat.concernStatus) {
        return cat.concernStatus.join(", ");
    } else if (carouselType === "unfed" && cat.lastFedTime) {
        return formatLastFed(cat.lastFedTime);
    }
    return "Unknown";
}

// Card Carousel used in Dashboard
const CardCarousel = ({cats, cardWidth, carouselType, ...card}) => {
    const navigation = useNavigation();
    const spaceBetweenCards = 24;

    return (
        <FlatList testID="card-carousel"
            style={{ height: cardWidth }}
            horizontal
            snapToAlignment="center"
            decelerationRate="normal"
            showsHorizontalScrollIndicator={false}
            scrollEventThrottle={4}
            snapToInterval={cardWidth + spaceBetweenCards}
            ItemSeparatorComponent={() => <View style={{width: spaceBetweenCards}} />}
            
            data={cats}
            keyExtractor={(item, index) => item.catID}
            renderItem={({item}) => {
                return (
                    <CatCard 
                        name={item.name}
                        photoURL={item.photoURLs[0]}
                        cardWidth={cardWidth}
                        onPress={() => navigation.navigate("catalogue", 
                            { screen: "CatProfile", initial: false, params: { catID: item.catID }})}
                        {...card}
                        info1={getInfo1(carouselType, item)}
                        info2={item.lastSeenLocation 
                            ? formatLastSeenSimple(item.lastSeenLocation, item.lastSeenTime)
                            : "Unknown"}
                    />
                );
            }}
        />
    );
};

export { getInfo1, CardCarousel };