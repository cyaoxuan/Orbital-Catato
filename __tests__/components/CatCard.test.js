import { CatCard, CatCardSimple } from "../../app/components/CatCard";
import { cleanup, fireEvent, render } from "@testing-library/react-native";

afterEach(cleanup);

describe("<CatCard />", () => {
    it("renders successfully", () => {
        const catCard = render(<CatCard />);
        expect(catCard).toBeDefined();
    });

    it("renders correctly", () => {
        const tree = render(<CatCard />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders the cover image with the correct photoURL", () => {
        const photoURL = require("../../assets/cats/cat-2-1.jpg");
        const { getByTestId } = render(<CatCard photoURL={photoURL} />);
        const cover = getByTestId("cover-image");
        expect(cover.props.source.uri).toBe(photoURL);
    });

    it("calls onPress callback when card action is pressed", () => {
        const profileOnPressMock = jest.fn();
        const locationOnPressMock = jest.fn();
        const { getAllByTestId } = render(
            <CatCard
                profileOnPress={profileOnPressMock}
                locationOnPress={locationOnPressMock}
            />
        );
        const buttons = getAllByTestId("button");
        fireEvent.press(buttons[0]);
        fireEvent.press(buttons[1]);
        expect(profileOnPressMock).toHaveBeenCalled();
        expect(locationOnPressMock).toHaveBeenCalled();
    });

    it("displays the name prop", () => {
        const name = "Kitty";
        const { queryByText } = render(<CatCard name={name} />);
        expect(queryByText(name)).not.toBeNull();
    });
});

describe("<CatCardSimple />", () => {
    it("renders successfully", () => {
        const catCardSimple = render(<CatCardSimple />);
        expect(catCardSimple).toBeDefined();
    });

    it("renders correctly", () => {
        const tree = render(<CatCardSimple />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders the cover image with the correct photoURL", () => {
        const photoURL = "https://picsum.photos/id/237/400/400";
        const { getByTestId } = render(<CatCardSimple photoURL={photoURL} />);
        const cover = getByTestId("cover-image");
        expect(cover.props.source.uri).toBe(photoURL);
    });

    it("calls onPress callback when card is pressed", () => {
        const onPressMock = jest.fn();
        const { getByTestId } = render(<CatCardSimple onPress={onPressMock} />);
        const cardContainerOuterLayer = getByTestId(
            "card-container-outer-layer"
        );
        const cardContainer = cardContainerOuterLayer.findByProps({
            testID: "card-container",
        });
        const card = cardContainer.findByProps({ testID: "card" });
        fireEvent.press(card);
        expect(onPressMock).toHaveBeenCalled();
    });

    it("displays the name prop", () => {
        const name = "Kitty";
        const { queryByText } = render(<CatCardSimple name={name} />);
        expect(queryByText(name)).not.toBeNull();
    });
});
