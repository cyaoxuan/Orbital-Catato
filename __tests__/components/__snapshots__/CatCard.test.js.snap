// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CatCard /> renders correctly 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "rgba(247, 243, 249, 1)",
      "borderRadius": 12,
      "height": 275,
      "margin": 4,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 3,
      "width": 300,
    }
  }
  testID="card-container-outer-layer"
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "rgba(247, 243, 249, 1)",
        "borderRadius": 12,
        "elevation": 1,
        "flex": 1,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 1,
      }
    }
    testID="card-container"
  >
    <View
      style={
        [
          {
            "flexShrink": 1,
          },
          undefined,
        ]
      }
      testID="card"
    >
      <View
        style={
          [
            {
              "backgroundColor": "#eeeeee",
              "height": 195,
              "overflow": "hidden",
            },
            {
              "borderRadius": 12,
            },
            {
              "height": 150,
              "resizeMode": "cover",
              "width": 300,
            },
          ]
        }
      >
        <Image
          accessibilityIgnoresInvertColors={true}
          siblings={
            [
              "Card.Cover",
              "Card.Title",
              "Card.Content",
            ]
          }
          source={1}
          style={
            [
              {
                "flex": 1,
                "height": undefined,
                "justifyContent": "flex-end",
                "padding": 16,
                "width": undefined,
              },
              {
                "borderRadius": 12,
              },
            ]
          }
          testID="cover-image"
        />
      </View>
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingLeft": 16,
            },
            {
              "minHeight": 50,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              [
                {
                  "fontFamily": "System",
                  "fontSize": 24,
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 32,
                },
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "minHeight": 30,
                    "paddingRight": 16,
                  },
                  {
                    "marginBottom": 2,
                  },
                  undefined,
                ],
              ]
            }
          >
            Name
          </Text>
        </View>
        <View />
      </View>
      <View
        style={
          [
            {
              "paddingHorizontal": 16,
            },
            {
              "paddingBottom": 16,
            },
            {
              "paddingBottom": 0,
              "paddingHorizontal": 4,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "flexDirection": "row",
              },
              undefined,
            ]
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": undefined,
                  "fontSize": 20,
                },
                {
                  "marginHorizontal": 4,
                },
                {
                  "fontFamily": "ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
            testID="icon"
          >
            
          </Text>
          <Text
            style={
              [
                {
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "letterSpacing": 0.15,
                  "lineHeight": 24,
                },
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                {
                  "flex": 1,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            Field: 
            <Text
              style={
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0.15,
                    "lineHeight": 24,
                  },
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  {
                    "flex": 1,
                  },
                ]
              }
            >
              Info
            </Text>
          </Text>
        </View>
        <View
          style={
            [
              {
                "flexDirection": "row",
              },
              undefined,
            ]
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": undefined,
                  "fontSize": 20,
                },
                {
                  "marginHorizontal": 4,
                },
                {
                  "fontFamily": "ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
            testID="icon"
          >
            
          </Text>
          <Text
            style={
              [
                {
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "letterSpacing": 0.15,
                  "lineHeight": 24,
                },
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                {
                  "flex": 1,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            Field: 
            <Text
              style={
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0.15,
                    "lineHeight": 24,
                  },
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  {
                    "flex": 1,
                  },
                ]
              }
            >
              Info
            </Text>
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`<CatCardSimple /> renders correctly 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "rgba(247, 243, 249, 1)",
      "borderRadius": 12,
      "flex": 0.5,
      "margin": 8,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 3,
    }
  }
  testID="card-container-outer-layer"
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "rgba(247, 243, 249, 1)",
        "borderRadius": 12,
        "elevation": 1,
        "flex": undefined,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 1,
      }
    }
    testID="card-container"
  >
    <View
      style={
        [
          {
            "flexShrink": 1,
          },
          undefined,
        ]
      }
      testID="card"
    >
      <View
        style={
          [
            {
              "backgroundColor": "#eeeeee",
              "height": 195,
              "overflow": "hidden",
            },
            {
              "borderRadius": 12,
            },
            {
              "height": NaN,
              "resizeMode": "cover",
              "width": undefined,
            },
          ]
        }
      >
        <Image
          accessibilityIgnoresInvertColors={true}
          siblings={
            [
              "Card.Cover",
              "Card.Title",
            ]
          }
          source={1}
          style={
            [
              {
                "flex": 1,
                "height": undefined,
                "justifyContent": "flex-end",
                "padding": 16,
                "width": undefined,
              },
              {
                "borderRadius": 12,
              },
            ]
          }
          testID="cover-image"
        />
      </View>
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingLeft": 16,
            },
            {
              "minHeight": 50,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              [
                {
                  "fontFamily": "System",
                  "fontSize": 28,
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 36,
                },
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "minHeight": 30,
                    "paddingRight": 16,
                  },
                  {
                    "marginBottom": 2,
                  },
                  undefined,
                ],
              ]
            }
          >
            Name
          </Text>
        </View>
        <View />
      </View>
    </View>
  </View>
</View>
`;
